conda/gateways/connection/download.py

```diff
@@ -35,118 +36,26 @@
+    # This complex exception translation strategy is reminiscent of def
+    # conda_http_errors(url, repodata_fn): in gateways/repodata
 
     try:
-        timeout = context.remote_connect_timeout_secs, context.remote_read_timeout_secs
-        session = get_session(url)
-        resp = session.get(url, stream=True, proxies=session.proxies, timeout=timeout)
-        if log.isEnabledFor(DEBUG):
-            log.debug(stringify(resp, content_max_len=256))
-        resp.raise_for_status()
-
-        content_length = int(resp.headers.get("Content-Length", 0))
-
-        # prefer sha256 over md5 when both are available
-        checksum_builder = checksum_type = checksum = None
-        if sha256:
-            checksum_builder = hashlib.new("sha256")
-            checksum_type = "sha256"
-            checksum = sha256
-        elif md5:
-            checksum_builder = hashlib.new("md5") if md5 else None
-            checksum_type = "md5"
-            checksum = md5
-
-        size_builder = 0
-        try:
-            with open(target_full_path, "wb") as fh:
-                streamed_bytes = 0
-                for chunk in resp.iter_content(2**14):
-                    # chunk could be the decompressed form of the real data
-                    # but we want the exact number of bytes read till now
-                    streamed_bytes = resp.raw.tell()
-                    try:
-                        fh.write(chunk)
-                    except OSError as e:
-                        message = (
-                            "Failed to write to %(target_path)s\n  errno: %(errno)d"
-                        )
-                        # TODO: make this CondaIOError
-                        raise CondaError(
-                            message, target_path=target_full_path, errno=e.errno
-                        )
-
-                    checksum_builder and checksum_builder.update(chunk)
-                    size_builder += len(chunk)
-
-                    if content_length and 0 <= streamed_bytes <= content_length:
-                        if progress_update_callback:
-                            progress_update_callback(streamed_bytes / content_length)
-
-            if content_length and streamed_bytes != content_length:
-                # TODO: needs to be a more-specific error type
-                message = dals(
-                    """
-                Downloaded bytes did not match Content-Length
-                  url: %(url)s
-                  target_path: %(target_path)s
-                  Content-Length: %(content_length)d
-                  downloaded bytes: %(downloaded_bytes)d
-                """
-                )
-                raise CondaError(
-                    message,
-                    url=url,
-                    target_path=target_full_path,
-                    content_length=content_length,
-                    downloaded_bytes=streamed_bytes,
-                )
+        yield
 
-        except OSError as e:
```

Comment:
PR moves this catch up a level into the parent try/except block and fixes hardcoded 104